app/support/reservations/validations.rb

```diff
@@ -21,10 +21,22 @@ module Reservations::Validations
 
     validate :starts_before_ends
     validate :duration_is_interval
+    validate :category_is_valid
   end
 
   # Validation Methods
 
+  def category_is_valid
+    return if category.blank?
+    if self.class.const_defined?(:CATEGORIES)
```

Comment:
> I'm suspecting that the reason for this complexity is so that we can add categories to admin reservations as well. Is that correct?

Yes, I wanted to leave the door open for admin categories. This also has to do with how we don't have STI completely for `Reservation` yet. If the model currently known as `Reservation` (that is, `type: nil`) could truly became `UserReservation`, I think it would help reduce the complexity of the validations. Or at least let us split the validations out into the 3 different types where we could.